src/app/tests/TestThreadBorderRouterManagementCluster.cpp

```diff
@@ -0,0 +1,247 @@
+    bool mInterfaceEnabled           = false;
+};
+
+constexpr chip::EndpointId sTestEndpointId    = 1;
+static TestDelegate * sTestDelegatePtr        = nullptr;
+static ServerInstance * sTestSeverInstancePtr = nullptr;
+
+class TestThreadBorderRouterManagementCluster : public ::testing::Test
+{
+public:
+    static void SetUpTestSuite()
+    {
+        sTestDelegatePtr = new ThreadBorderRouterManagement::TestDelegate();
+        ASSERT_TRUE(sTestDelegatePtr);
+        sTestSeverInstancePtr = new ThreadBorderRouterManagement::ServerInstance(sTestEndpointId, sTestDelegatePtr);
+        ASSERT_TRUE(sTestSeverInstancePtr);
+    }
+
+    static void TearDownTestSuite()
+    {
+        delete sTestDelegatePtr;
+        delete sTestSeverInstancePtr;
+    }
+
+    void TestAttributeRead();
+    void TestCommandHandle();
+};
+
+// Test ReadXX functions in ThreadBorderRouterManagement ServerInstance
+TEST_F_FROM_FIXTURE(TestThreadBorderRouterManagementCluster, TestAttributeRead)
+{
+    BitFlags<Feature> featureMap;
+    EXPECT_EQ(sTestSeverInstancePtr->ReadFeatureMap(featureMap), CHIP_NO_ERROR);
+    EXPECT_TRUE(featureMap.Has(Feature::kPANChange));
+    sTestDelegatePtr->mPanChangeSupported = false;
+    featureMap.ClearAll();
+    EXPECT_EQ(sTestSeverInstancePtr->ReadFeatureMap(featureMap), CHIP_NO_ERROR);
+    EXPECT_FALSE(featureMap.Has(Feature::kPANChange));
+    char borderRouterName[kBorderRouterNameMaxLength + 10];
+    MutableCharSpan nameSpan = MutableCharSpan(borderRouterName);
+    EXPECT_EQ(sTestSeverInstancePtr->ReadBorderRouterName(nameSpan), CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    nameSpan                                      = MutableCharSpan(borderRouterName);
+    sTestDelegatePtr->mUseInvalidBorderRouterName = false;
+    EXPECT_EQ(sTestSeverInstancePtr->ReadBorderRouterName(nameSpan), CHIP_NO_ERROR);
+    EXPECT_TRUE(nameSpan.data_equal(CharSpan("TestName", strlen("TestName"))));
+    uint8_t borderAgentId[kBorderAgentIdLength];
+    MutableByteSpan agentIdSpan = MutableByteSpan(borderAgentId);
+    EXPECT_EQ(sTestSeverInstancePtr->ReadBorderAgentID(agentIdSpan), CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    agentIdSpan                             = MutableByteSpan(borderAgentId);
+    sTestDelegatePtr->mTestBorderAgentIdLen = kBorderAgentIdLength;
+    EXPECT_EQ(sTestSeverInstancePtr->ReadBorderAgentID(agentIdSpan), CHIP_NO_ERROR);
+    EXPECT_TRUE(agentIdSpan.data_equal(ByteSpan(sTestDelegatePtr->kTestBorderAgentId)));
+    uint16_t threadVersion;
+    EXPECT_EQ(sTestSeverInstancePtr->ReadThreadVersion(threadVersion), CHIP_NO_ERROR);
+    EXPECT_EQ(threadVersion, sTestDelegatePtr->kTestThreadVersion);
+    bool interfaceEnabled;
+    EXPECT_EQ(sTestSeverInstancePtr->ReadInterfaceEnabled(interfaceEnabled), CHIP_NO_ERROR);
+    EXPECT_FALSE(interfaceEnabled);
+    Optional<uint64_t> timestamp;
+    EXPECT_EQ(sTestSeverInstancePtr->ReadActiveDatasetTimestamp(timestamp), CHIP_NO_ERROR);
+    EXPECT_FALSE(timestamp.HasValue());
```

Comment:
Please make oen separate scope per attribute. This is very hard to read.

Also, initialize all stack variables to guard against uninitialized usage.

Reply:
Done